applications/crossbar/doc/port_requests.md

```diff
@@ -1,262 +1,1030 @@
+        }
+    },
+    "request_id": "ed71507eeccbd3eeab8ccd7bd6cf023c",
+    "revision": "6-ade05f2d7e539be79f425bfb39e41cef",
+    "status": "success"
+}
+```
+
+
+#### Add an attachment to a port request
+
+Note: if `ATTACHMENT_ID` does not end with `.pdf` the extension will be appended.
+
+> PUT /v2/accounts/{ACCOUNT_ID}/port_requests/{PORTREQUEST_ID}/attachments
+
+```shell
+curl -v -X PUT \
+    -H "X-Auth-Token: {AUTH_TOKEN}" \
+    -H "Content-Type: application/pdf" \
+    --data-binary @/path/to/file.pdf \
+    http://{SERVER}:8000/v2/accounts/{ACCOUNT_ID}/port_requests/{PORTREQUEST_ID}/attachments?filename={ATTACHMENT_ID}'
+```
+
+```json
+{
+    "auth_token": "{AUTH_TOKEN}",
+    "data": {},
+    "request_id": "e9db1aa74945884eed5a4545a405b490",
+    "revision": "2-7ab1daee08211c782a6cdc5425886be4",
+    "status": "success"
+}
+```
+
+
+#### Get an attachment from a port request
+
+> GET /v2/accounts/{ACCOUNT_ID}/port_requests/{PORTREQUEST_ID}/attachments/{ATTACHMENT_ID}
+
+```shell
+curl -v -X GET \
+    -H "X-Auth-Token: {AUTH_TOKEN}" \
+    -H "Accept: application/pdf" \
+    http://{SERVER}:8000/v2/accounts/{ACCOUNT_ID}/port_requests/{PORTREQUEST_ID}/attachments/{ATTACHMENT_ID}
+```
+
+Streams back the contents of the PDF file `{ATTACHMENT_ID}`.
+
+
+#### Replace an attachment on a port request
+
+> POST /v2/accounts/{ACCOUNT_ID}/port_requests/{PORTREQUEST_ID}/attachments/{ATTACHMENT_ID}
+
+```shell
+curl -v -X POST \
+    -H "X-Auth-Token: {AUTH_TOKEN}" \
+    -H "Content-Type: application/pdf" \
+    --data-binary @/path/to/file.pdf \
+    http://{SERVER}:8000/v2/accounts/{ACCOUNT_ID}/port_requests/{PORTREQUEST_ID}/attachments/{ATTACHMENT_ID}
+```
+
+```json
+{
+    "auth_token": "{AUTH_TOKEN}",
+    "data": {},
+    "request_id": "5ccaf7eb473c350c2e460ae5b4a76697",
+    "revision": "4-288d2d9e121c8e5d5e0d137626d70536",
+    "status": "success"
+}
+```
+
+
+#### Delete an attachment on a port request
+
+> DELETE /v2/accounts/{ACCOUNT_ID}/port_requests/{PORTREQUEST_ID}/attachments/{ATTACHMENT_ID}
+
+```shell
+curl -v -X DELETE \
+    -H "X-Auth-Token: {AUTH_TOKEN}" \
+    http://{SERVER}:8000/v2/accounts/{ACCOUNT_ID}/port_requests/{PORTREQUEST_ID}/attachments/{ATTACHMENT_ID}
+```
+
+```json
+{
+    "auth_token": "{AUTH_TOKEN}",
+    "data": {},
+    "request_id": "1178da494ef5a953240b33eeaa7df859",
+    "revision": "undefined",
+    "status": "success"
+}
+```
+
+
+#### Indicate a port is ready to be processed
+
+> PATCH /v2/accounts/{ACCOUNT_ID}/port_requests/{PORTREQUEST_ID}/submitted
+
+```shell
+curl -v -X PATCH \
+    -H "X-Auth-Token: {AUTH_TOKEN}" \
+    http://{SERVER}:8000/v2/accounts/{ACCOUNT_ID}/port_requests/{PORTREQUEST_ID}/submitted
+```
+
+##### Success
+
+```json
+{
+    "auth_token": "{AUTH_TOKEN}",
+    "data": {
+        "created": 63630107130,
+        "id": "{PORTREQUEST_ID}",
+        "name": "Porting 202.555.9000",
         "numbers": {
-            "{{NUMBER}}": {}
+            "+12025559000": {}
```

Comment:
Hm my rationale is: there is no way to know what will be here from the request alone, so data should stay intact here.